cluster/v1beta1/helpers.go

```diff
@@ -217,83 +234,41 @@ func (pdct *PlacementDecisionClustersTracker) generateGroupsNameIndex() {
 	}
 }
 
-// Existing() returns the tracker's existing decision cluster names of groups listed in groupKeys.
+// ExistingClusterGroups returns the tracker's existing decision cluster groups for groups listed in groupKeys.
 // Return empty set when groupKeys is empty.
-func (pdct *PlacementDecisionClustersTracker) Existing(groupKeys []GroupKey) (sets.Set[string], []GroupKey, map[GroupKey]sets.Set[string]) {
-	existingGroupKeys, existingClusterGroups := pdct.ExistingClusterGroups(groupKeys)
-
-	existingClusters := sets.New[string]()
-	for _, clusterSets := range existingClusterGroups {
-		existingClusters.Insert(clusterSets.UnsortedList()...)
-	}
-
-	return existingClusters, existingGroupKeys, existingClusterGroups
-}
-
-// ExistingBesides returns the tracker's existing decision cluster names except cluster groups listed in groupKeys.
-// Return all the clusters when groupKeys is empty.
-func (pdct *PlacementDecisionClustersTracker) ExistingBesides(groupKeys []GroupKey) (sets.Set[string], []GroupKey, map[GroupKey]sets.Set[string]) {
-	existingGroupKeys, existingClusterGroups := pdct.ExistingClusterGroupsBesides(groupKeys)
-
-	existingClusters := sets.New[string]()
-	for _, clusterSets := range existingClusterGroups {
-		existingClusters.Insert(clusterSets.UnsortedList()...)
-	}
-
-	return existingClusters, existingGroupKeys, existingClusterGroups
-}
-
-// ExistingClusterGroups returns the tracker's existing decision cluster names for groups listed in groupKeys.
-// Return empty set when groupKeys is empty.
-func (pdct *PlacementDecisionClustersTracker) ExistingClusterGroups(groupKeys []GroupKey) ([]GroupKey, map[GroupKey]sets.Set[string]) {
+func (pdct *PlacementDecisionClustersTracker) ExistingClusterGroups(groupKeys []GroupKey) ClusterGroupsMap {
```

Comment:
can we use ExistingClusterGroups(groupKeys ...GroupKey)?

Reply:
Have updated code.